docs/specs/delete-user.md

```diff
@@ -0,0 +1,23 @@
+# Delete User
+
+  * [Deleting user](#deleting-user)
+  * [Cached data](#cached-data)
+  * [Future works](#future-works)
+
+## Deleting user
+
+A user can be deleted by admins. A deleted user would have all its associated
+data (exposed from API) erased from database (hard-delete), including
+identities, authenticators, password data, sessions, etc. 
+
+## Cached data
+
+Some internal data may be present in cache (Redis), such as OAuth states,
+MFA device tokens, rate limit counter. These data would remain in the cache
+until its natural expiry.
+
+## Future works
+
+- Web-hooks
+- Reserve deleted login ID
+- Self-service deletion
```

Comment:
A common feature needed is "Soft delete for N days and delete", I think it sort of like "Disable for N days and then delete" in our case. Maybe put it in future work so it is considered? (Although it should be relatively trivial)

Reply:
BTW, same applies to self-service deleteion too, for a lot of consumer apps the Business would prefer to let the user delete but have a 30 days period to revert its decision. In those case we need to display a better default copywriting of reason for disable users for this case.